### ISSUE TITLE
Override not setup properly (possibly due to mount/symlinks)

### ISSUE THREAD

**Issue by gsingh93** [brson/multirust#40]
I ran `multirust override beta` in a directory and the beta was installed. Running `multirust list-overrides` showed the current directory (let's say `/foo`) as overriden. However, running `rustc --version` showed `rustc` was still the nightly version. I ran the override command again and this time `rustc --version` showed the proper version, but now `multirust list-overrides` shows two overrides, `/foo` and `/mnt/foo`. In my setup, `/foo` is a symlink to `/mnt/foo`.

**Comment by brson** [comment]
Thanks for the report. I'll investigate sometime next cycle. Need to do a number of multirust upgrades.

**Comment by brson** [comment]
I haven't been able to reproduce this on Linux. I set my default to nightly, then ran `multirust override beta` in a directory, `./foo` that was a symlink to another one, `./mnt/foo`, and subsequest `multirust show-override` and  `rustc --version` from `./foo` looked correct. Invocations from `./mnt/foo` did not detect the override, which I expect.

Are you on OS X perhaps? Does the test suite, `git submodule update --recursive --init && ./test.sh` pass for you?

**Comment by gsingh93** [comment]
I'm using Arch Linux. I just tested this again and I'm having the same issue, but I'm not completely sure this is a problem with multirust. I'm assuming multirust uses `pwd` to get the current directory? My `pwd` starts of showing the directory without the `/mnt`, and at some point (I'm trying to determine at what point that is), it switches over to showing the current directory with the `/mnt`. So I think this might be a general Linux/mount/pwd issue.

**Comment by gsingh93** [comment]
Maybe `multirust` can use `pwd -P` to avoid the issue all together? I don't know if that causes portability problems though.

**Comment by brson** [comment]
Yeah, it uses `pwd`. Does replacing `pwd` with `pwd -P` definitely work for you? `pwd -P` exists on OS X at least.

**Comment by gsingh93** [comment]
I figured out why sometimes `pwd` shows the full directory with the `/mnt` and sometimes it doesn't.

I use `terminator` for my terminal, but this behavior could actually apply to any software that opens a terminal. When I split the terminal (ctrl+shift+o), terminator automatically `cd`s into the directory you were just in. For some reason, it `cd`s into the directory `pwd -P` instead of `pwd`, which includes the `/mnt`.

So using `pwd -P` will fix this problem for me, for anyone who uses software that does the same thing as `terminator`, and for anyone who sometimes `cd`s into a directory with a symlink and sometimes (for whatever reason) uses the full physical path.

However, using `pwd -P` now will break overrides for anyone who already has an override in a symlinked directory, as the result of `pwd -P` will differ from `pwd` in that directory. If you don't want to make this change because of that, I don't mind, I'll just deal with this problem when it happens.

**Comment by gsingh93** [comment]
@brson Feel free to close this if you don't think it needs to be fixed.

**Comment by brson** [comment]
@gsingh93 Does https://github.com/brson/multirust/pull/81 fix your problem?
